### PR TITLE
Improve rpc framework

### PR DESCRIPTION
**Pull request by realwanpengli**
1. Decouple Rpc master/slave from user plugin
2. Finish simple handler to handle pipeline in Rpc framework
3. Make plugin able to install via its name

### CODE REVIEW

File: SignalRServiceBenchmarkPlugin/framework/master/Program.cs [hunk 1]
```diff
@@ -7,12 +7,14 @@
...
 using System.IO;
 using Common;
+using Plugin.Base;
 
 namespace Rpc.Master
 {
     class Program
     {
         private static readonly int _maxRertryConnect = 100;
+        private static IPugin _plugin;
```
Comment (vwxyzh) [inline SignalRServiceBenchmarkPlugin/framework/master/Program.cs]: typo

Comment (realwanpengli) [reply SignalRServiceBenchmarkPlugin/framework/master/Program.cs]: Got it , thanks...

File: SignalRServiceBenchmarkPlugin/framework/master/Program.cs [hunk 2]
```diff
@@ -22,58 +24,99 @@ static async Task Main(string[] args)
...
+            // Check whether the plugin installed
+            var success = true;
+            foreach (var task in tasks)
             {
-                channels.Add(new Channel($"{slaveList[i]}:{rpcPort}", ChannelCredentials.Insecure,
-                    new ChannelOption[] {
-                        // For Group, the received message size is very large, so here set 8000k
-                        new ChannelOption(ChannelOptions.MaxReceiveMessageLength, 8192000)
-                    }));
+                if (!task.GetAwaiter().GetResult()) success = false;
```
[the remark this answers is not in the document]
Comment (realwanpengli) [reply SignalRServiceBenchmarkPlugin/framework/master/Program.cs]: Changed `task.GetAwaiter().GetResult()` to `task.Result`

File: SignalRServiceBenchmarkPlugin/framework/master/Program.cs [hunk 3]
```diff
@@ -22,58 +22,99 @@ static async Task Main(string[] args)
...
+            Log.Information($"Install plugin in slaves...");
+
+            var tasks = new List<Task<bool>>();
+
+            // Try to install plugin
+            foreach (var client in clients)
+            {
+                tasks.Add(client.InstallPluginAsync(moduleName));
+            }
+            await Task.WhenAll(tasks);
```
Comment (vwxyzh) [inline SignalRServiceBenchmarkPlugin/framework/master/Program.cs]: considering: `await Task.WhenAll(from client in clients select client.InstallPluginAsync(moduleName);`

Comment (realwanpengli) [reply SignalRServiceBenchmarkPlugin/framework/master/Program.cs]: Got it, thanks

**Pull request merged**